src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -418,6 +525,36 @@ private static bool CheckBytesInAsciiRange(Vector<sbyte> check)
             return Vector.GreaterThanAll(check, Vector<sbyte>.Zero);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool CheckBytesInAsciiRange(Vector256<sbyte> check, Vector256<sbyte> zero)
+        {
+            if (Avx2.IsSupported)
+            {
+                var mask = Avx2.CompareGreaterThan(check, zero);
+
+                return (uint)Avx2.MoveMask(mask) == 0xFFFF_FFFF;
+            }
+            else
+            {
+                throw new PlatformNotSupportedException("should not be here");
```

Comment:
Will never be reached, but to indiciate that if Arm / Neon comes up as intrinsics the check may be expanded here. JIT eleminates this branch anyway.